File: Event/src/main/java/com/esprit/event/DAO/entities/Event.java
package com.esprit.event.DAO.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;


@Entity
@Getter
@Setter
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idEvent;
    private String eventName;
    private String eventDescription;
    private LocalDateTime eventDate;
    @Enumerated(EnumType.STRING)
    private Category eventCategory;
    @ManyToOne
    private Centre centre;

    @ManyToMany
    private List<User> participants;

    @ManyToOne
    private User eventCreator;
}

File: Front_end/src/app/models/Event.ts
import { CategoryEnum } from './CategoryEnum';
import { Centre } from './Centre';
import { MainCategory } from './MainCategory';
import { User } from './User';

export class Event {
  idEvent: number;
  eventName: string;
  eventDescription: string;
  eventDate: string; // Use string because Angular handles date conversion
  eventCategory: CategoryEnum;
  mainCategory: MainCategory;
  imageUrl:  string;
  centre: number;
  participants: number[];
  eventCreator: number;
  eventDateOnly?: string;
  eventTimeOnly?: string;
  eventCreatorName?: string;
  isExpanded: boolean;
}
